src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py

```diff
@@ -6,64 +6,196 @@
+        This parameter takes precedence over any value specified for the
+        augmentation_mode.
+
+    flip_vertical : bool optional
+        Boolean flag that indicates whether the object can be flipped
+        vertically,
+        i.e. whether a mirror image of the object should appear can be in the
+        augmented data.
+        This will probably be false for logos.
+        If the user wants to set this field and has multiple starter images in
+        the dataset, they can also provide values for this field as a different
+        column in the dataset.
+        Defaults to True.
+        If 'dataset' is an SFrame, the flip_vertical value specified as this
+        parameter will be applied to all the images provided in the SFrame.
+        If you'd like different images in the 'dataset' SFrame to have different
+        flip_vertical's, those values can be specified as a column in the
+        SFrame, and that column name should be passed in to the
+        'flip_vertical' parameter
+        This parameter takes precedence over any value specified for the
+        augmentation_mode.
+
+    backgrounds : optional SArray
+        A list of backgrounds that the user wants to provide for data
+        augmentation.
+        If this is provided, only the backgrounds provided as this field will be
+        used for augmentation.
+
+    batch_size : int
+        The number of images per training iteration. If 0, then it will be
+        automatically determined based on resource availability.
+
+    max_iterations : int
+        The number of training iterations. If 0, then it will be automatically
+        be determined based on the amount of data you provide.
+
+    verbose : bool optional
+        If True, print progress updates and model details.
+
+    **kwargs : dict optional
+        A dictionary of advanced parameters.
+        Here are the parameters currently supported:
+        - yaw  : rotation of the logo (or any 2-D starter image) along Y axis
+        - pitch: rotation of the logo (or any 2-D starter image) along X axis
+        - roll : rotation of the logo (or any 2-D starter image) along Z axis
+            A list of the ranges that the user wants the respective angle 
+            parameter to be in. e.g. yaw = [(0, 90), (180, 270)]. 
+            This would assume a normal distribution by default.
+            Here is some ASCII art explaining yaw, pitch, roll
+
+            ##############################      y (yaw)
+            ##############################      ^
+            ########## ________ ##########      |
+            ##########|  logo  |##########      |
+            ##########|________|##########      |-------> x (pitch)
+            ##############################     /
+            ##############################    /
+            ##############################   /
+            ##############################  z (roll)
+
+        - blur: A boolean flag indicating whether augmented images should be
```

Comment:
Maybe `blur` and `lighting_perturbation` should be an ints rather than a bools. Does it really have to be all or nothing?

Reply:
I think we should think twice before publishing API for these kinds of augmentations that occur after the superimposition of the user image over the background. What happens when we expose the augmentation options in OD? How would these interact?

Note that OD users can already manipulate e.g. jitter in hue, brightness, and saturation, as well as noise, via the `kwargs` argument of `turicreate.object_detector.create`.

Reply:
I didn't know those things could be manipulated via the `kwargs` of argument of `turicreate.object_detector.create`. Looks like we haven't documented that: https://apple.github.io/turicreate/docs/api/generated/turicreate.object_detector.create.html#turicreate.object_detector.create

I created #1901 to track this issue.

Reply:
Yeah, I suspect the intent was for it to be "hidden" API. Before we promoted the `batch_size` parameter to the documented API, for example, we did tell some users how to manipulate the batch size in this manner.

Ultimately, we should decide on some consistent API for data augmentation that fits well with the relationship between OD and one-shot OD. In the absence of that design, I think we should err on the side of publishing the minimal API that serves users

Reply:
Let's table exposing blur, color_perturbation, and noise until we have a story around how they interact with OD as well? Does that sound good?